protocol/x/clob/keeper/clob_pair.go

```diff
@@ -595,6 +595,21 @@ func (k Keeper) validateInternalOperationAgainstClobPairStatus(
 			clobPairId,
 			types.ClobPair_STATUS_INITIALIZING,
 		)
+	case types.ClobPair_STATUS_FINAL_SETTLEMENT:
+		// Only allow deleveraging events. This allows the protocol to close out open
+		// positions in the market. All trading is blocked.
+		if match := internalOperation.GetMatch(); match != nil {
+			if match.GetMatchPerpetualDeleveraging() != nil {
```

Comment:
Suggestion: cleaner to put on one line? Up to you

```suggestion
		if match := internalOperation.GetMatch(); match != nil && match.GetMatchPerpetualDeleveraging() != nil {
```